syntax = "proto3";
package oceanim;

option cc_generic_services =true;

import "common/messages.proto";

// 编译命令
// protoc --proto_path=/workspaces/CPP_Server/ocean_im --cpp_out=/workspaces/CPP_Server/ocean_im server/logic/logic.proto

service LogicService{
    rpc Test(Ping) returns (Pong);

    rpc SendMsg(NewMsg) returns (MsgReply);
    rpc PullData(Ping) returns (Msgs);

    rpc GetMsgs(MsgIdRange) returns (Msgs);
    rpc GetFriends(UserId) returns (UserInfos);
    rpc GetGroups(UserId) returns (GroupInfos);
    rpc GetGroupMembers(GroupId) returns (UserInfos);
}
